puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -30,12 +33,12 @@ def save(request)
           # when we attempt to use ActiveSupport 2.3.16 on RHEL 5 with
           # legacy storeconfigs.
           "environment" => request.options[:environment] || request.environment.to_s,
-          "producer_timestamp" => request.options[:producer_timestamp] || Time.now.iso8601(5),
+          "producer_timestamp" => Puppet::Util::Puppetdb.to_wire_time(current_time),
           "producer" => Puppet[:node_name_value]
         }
       end
 
-      submit_command(request.key, payload, CommandReplaceFacts, 5)
+      submit_command(request.key, payload, CommandReplaceFacts, 5, current_time.utc)
```

Comment:
It seems strange to me that we are sending non-UTC times to PDB. We just convert it to UTC over there anyway. I didn't want to change that in this PR, but it seems like we should make that change.

Are you suggesting I clone the time instance so that line 36 and 41 are not order dependent?